### PR TITLE
Add drf logging to API ViewSets

### PR DESCRIPTION
**Pull request by mayankkapoor**
This now fixes the issue with MySQL backend. The fix was implemented here: https://github.com/mayankkapoor/drf-tracking/commit/e832621b23314529e3809b320d2742719278fef4

### CODE REVIEW

**Comment by sorenisanerd** [comment]
This only adds it to MirrorViewSet. Can you add it everywhere, please?

Since all of our ViewSets share their baseclasses, perhaps create an AasembleViewSet that all the rest can inherit from?

**Comment by sorenisanerd** [comment]
@mayankkapoor Sweet, MySQL testing has landed. Can you address my last comment here, so we can land this, too?

Thanks! This is an excellent idea.

**Comment by mayankkapoor** [comment]
@sorenh Made the changes as asked. Almost there, but facing 1 error related to db transactions. 

```
ERROR: test_create_duplicate_repository (aasemble.django.apps.api.tests.APIv1Tests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/python-aasemble.django/aasemble/django/apps/api/tests.py", line 117, in test_create_duplicate_repository
    response = self.client.post(self.repository_list_url, data, format='json')
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/rest_framework/test.py", line 170, in post
    path, data=data, format=format, content_type=content_type, **extra)
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/rest_framework/test.py", line 92, in post
    return self.generic('POST', path, data, content_type, **extra)
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/django/test/client.py", line 379, in generic
    return self.request(**r)
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/rest_framework/test.py", line 159, in request
    return super(APIClient, self).request(**kwargs)
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/rest_framework/test.py", line 111, in request
    request = super(APIRequestFactory, self).request(**kwargs)
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/django/test/client.py", line 466, in request
    six.reraise(*exc_info)
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/rest_framework/viewsets.py", line 87, in view
    return self.dispatch(request, *args, **kwargs)
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/rest_framework/views.py", line 468, in dispatch
    self.response = self.finalize_response(request, response, *args, **kwargs)
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/rest_framework_tracking/mixins.py", line 58, in finalize_response
    request.log.save()
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/django/db/models/base.py", line 734, in save
    force_update=force_update, update_fields=update_fields)
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/django/db/models/base.py", line 762, in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/django/db/models/base.py", line 827, in _save_table
    forced_update)
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/django/db/models/base.py", line 877, in _do_update
    return filtered._update(values) > 0
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/django/db/models/query.py", line 580, in _update
    return query.get_compiler(self.db).execute_sql(CURSOR)
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 1062, in execute_sql
    cursor = super(SQLUpdateCompiler, self).execute_sql(result_type)
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 840, in execute_sql
    cursor.execute(sql, params)
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/django/db/backends/utils.py", line 59, in execute
    self.db.validate_no_broken_transaction()
  File "/Users/mayankkapoor/Dropbox/01Jio/02code/aaSemble_django/env/lib/python2.7/site-packages/django/db/backends/base/base.py", line 327, in validate_no_broken_transaction
    "An error occurred in the current transaction. You can't "
TransactionManagementError: An error occurred in the current transaction. You can't execute queries until the end of the 'atomic' block.
```

Related issue fix on stackoverflow: http://stackoverflow.com/questions/21458387/transactionmanagementerror-you-cant-execute-queries-until-the-end-of-the-atom.

Investigating. Any advice?

**Comment by sorenisanerd** [comment]
I haven't entirely wrapped my head around the details of the failure, but the answer on Stackoverflow sounds very plausible. Have you tried wrapping that operation in a transaction.atomic()?

**Comment by mayankkapoor** [comment]
@sorenh I did try to wrap the test case statements in transaction.atomic(), and also the request.log.save() statement in drf-tracking as well. No luck. Will look at it again later.

**Comment by mayankkapoor** [comment]
@sorenh Ready for review. CircleCI failed due to using a cached virtualenv.

File: aasemble/django/apps/api/tests.py [hunk 1]
```diff
@@ -244,9 +244,7 @@ def test_create_same_name_repository_different_user(self):
 
     def test_fetch_builds(self):
         authenticate(self.client, 'eric')
-        # 3 queries: Authenticate, count results, fetch results
-        with self.assertNumQueries(3):
-            response = self.client.get(self.build_list_url)
+        response = self.client.get(self.build_list_url)
```
Comment (sorenisanerd) [inline aasemble/django/apps/api/tests.py]: This makes me cry :)

Why can't we keep these checks?

**Comment by mayankkapoor** [comment]
@sorenh ready for review.

**Comment by sorenisanerd** [comment]
Looks good. I'll merge a bit later. This is because when I move this to production, I need to add drf-tracking to my installed_apps before this lands, otherwise it'll all fail.

In the future, we should probably have a check to see if drf_tracking is in INSTALLED_APPS, and if not, not add the Mixin. That way, it's easy to disable: Just remove it from INSTALLED_APPS.

**Pull request merged**